gdsfactory/component.py

```diff
@@ -2596,8 +2596,29 @@ def metadata(self) -> dict:
         )
         return dict(self.settings)
 
+    def __reduce__(self):
+        """Gdstk Cells cannot be directly pickled. This method overrides binary serialization with GDS serialization."""
+        return deserialize_gds, serialize_gds(self)
+
 
 # Component functions
+def serialize_gds(component):
+    """Saves Component as GDS + YAML metadata in temporary files with unique name."""
+    gds_filepath = GDSDIR_TEMP / component.name
```

Comment:
@joamatab should this use something else than the name to guarantee there are no collisions?

Reply:
I think name is the best we can use :)